### ISSUE TITLE
Infinite "Tried to load PackageNames from disc, re-caching" loop

### ISSUE THREAD

**Issue by non-Jedi** [julia-vscode/SymbolServer.jl#57]
After caches were finally created for the packages in #56, SymbolServer entered an infinite loop of `[ Info: Tried to load LinearAlgebra but failed to load from disc, re-caching.` with a bunch of different packages. Logs before I <kbd><kbd>Ctrl</kbd>-<kbd>c</kbd></kbd>ed it:

```
julia> using LanguageServer

julia> server = LanguageServer.LanguageServerInstance(stdin, stdout, false, "/home/adam/repos/ChemicalEngineeringToolbox.jl/", "", Dict())
Root: Set(String[])

julia> run(server)
[ Info: Started symbol server
[ Info: Unitful not stored on disc
[ Info: StaticArrays not stored on disc
[ Info: Tried to load Statistics but failed to load from disc, re-caching.
[ Info: UnitfulUS not stored on disc
[ Info: DelimitedFiles not stored on disc
[ Info: ModelingToolkit not stored on disc
[ Info: Interpolations not stored on disc
[ Info: Tried to load LinearAlgebra but failed to load from disc, re-caching.
[ Info: Tried to load Unitful but failed to load from disc, re-caching.
[ Info: Tried to load Serialization but failed to load from disc, re-caching.
[ Info: Tried to load Random but failed to load from disc, re-caching.
[ Info: Tried to load Libdl but failed to load from disc, re-caching.
[ Info: Tried to load LinearAlgebra but failed to load from disc, re-caching.
[ Info: Tried to load Unitful but failed to load from disc, re-caching.
[ Info: Tried to load Statistics but failed to load from disc, re-caching.
[ Info: Tried to load Serialization but failed to load from disc, re-caching.
[ Info: Tried to load Random but failed to load from disc, re-caching.
[ Info: Tried to load SparseArrays but failed to load from disc, re-caching.
[ Info: Tried to load Libdl but failed to load from disc, re-caching.
[ Info: Tried to load StaticArrays but failed to load from disc, re-caching.
[ Info: Tried to load LinearAlgebra but failed to load from disc, re-caching.
[ Info: Tried to load Unitful but failed to load from disc, re-caching.
[ Info: Tried to load Statistics but failed to load from disc, re-caching.
[ Info: Tried to load Serialization but failed to load from disc, re-caching.
[ Info: Tried to load Random but failed to load from disc, re-caching.
[ Info: Tried to load SparseArrays but failed to load from disc, re-caching.
[ Info: Tried to load Libdl but failed to load from disc, re-caching.
[ Info: Tried to load StaticArrays but failed to load from disc, re-caching.
[ Info: Tried to load UnitfulUS but failed to load from disc, re-caching.
[ Info: Tried to load Logging but failed to load from disc, re-caching.
[ Info: Tried to load SparseArrays but failed to load from disc, re-caching.
[ Info: Tried to load Sockets but failed to load from disc, re-caching.
[ Info: Tried to load Random but failed to load from disc, re-caching.
[ Info: Tried to load InteractiveUtils but failed to load from disc, re-caching.
[ Info: Tried to load Libdl but failed to load from disc, re-caching.
[ Info: Tried to load Markdown but failed to load from disc, re-caching.
[ Info: Tried to load Unitful but failed to load from disc, re-caching.
[ Info: Tried to load Distributed but failed to load from disc, re-caching.
[ Info: Tried to load Serialization but failed to load from disc, re-caching.
[ Info: Tried to load Test but failed to load from disc, re-caching.
[ Info: Tried to load StaticArrays but failed to load from disc, re-caching.
[ Info: Tried to load LinearAlgebra but failed to load from disc, re-caching.
[ Info: Tried to load Statistics but failed to load from disc, re-caching.
[ Info: Tried to load Base64 but failed to load from disc, re-caching.
[ Info: Tried to load UnitfulUS but failed to load from disc, re-caching.
[ Info: Tried to load Logging but failed to load from disc, re-caching.
[ Info: Tried to load SparseArrays but failed to load from disc, re-caching.
[ Info: Tried to load Sockets but failed to load from disc, re-caching.
[ Info: Tried to load Random but failed to load from disc, re-caching.
[ Info: Tried to load InteractiveUtils but failed to load from disc, re-caching.
[ Info: Tried to load DelimitedFiles but failed to load from disc, re-caching.
[ Info: Tried to load Libdl but failed to load from disc, re-caching.
[ Info: Tried to load Markdown but failed to load from disc, re-caching.
[ Info: Tried to load Unitful but failed to load from disc, re-caching.
[ Info: Tried to load Distributed but failed to load from disc, re-caching.
[ Info: Tried to load Serialization but failed to load from disc, re-caching.
[ Info: Tried to load Test but failed to load from disc, re-caching.
[ Info: Tried to load StaticArrays but failed to load from disc, re-caching.
[ Info: Tried to load LinearAlgebra but failed to load from disc, re-caching.
[ Info: Tried to load Statistics but failed to load from disc, re-caching.
[ Info: Tried to load Mmap but failed to load from disc, re-caching.
[ Info: Tried to load Base64 but failed to load from disc, re-caching.
[ Info: Tried to load Logging but failed to load from disc, re-caching.
[ Info: Tried to load SparseArrays but failed to load from disc, re-caching.
[ Info: Tried to load MuladdMacro but failed to load from disc, re-caching.
[ Info: Tried to load NaNMath but failed to load from disc, re-caching.
[ Info: Tried to load Random but failed to load from disc, re-caching.
[ Info: Tried to load DelimitedFiles but failed to load from disc, re-caching.
[ Info: Tried to load DataStructures but failed to load from disc, re-caching.
[ Info: Tried to load BinDeps but failed to load from disc, re-caching.
[ Info: Tried to load Unitful but failed to load from disc, re-caching.
[ Info: Tried to load Compat but failed to load from disc, re-caching.
[ Info: Tried to load URIParser but failed to load from disc, re-caching.
[ Info: Tried to load CanonicalTraits but failed to load from disc, re-caching.
[ Info: Tried to load DiffEqBase but failed to load from disc, re-caching.
[ Info: Tried to load ArrayInterface but failed to load from disc, re-caching.
[ Info: Tried to load LinearAlgebra but failed to load from disc, re-caching.
[ Info: Tried to load Pkg but failed to load from disc, re-caching.
[ Info: Tried to load Printf but failed to load from disc, re-caching.
[ Info: Tried to load DiffEqDiffTools but failed to load from disc, re-caching.
[ Info: Tried to load UnitfulUS but failed to load from disc, re-caching.
[ Info: Tried to load TableTraits but failed to load from disc, re-caching.
[ Info: Tried to load IteratorInterfaceExtensions but failed to load from disc, re-caching.
[ Info: Tried to load Sockets but failed to load from disc, re-caching.
[ Info: Tried to load Dates but failed to load from disc, re-caching.
[ Info: Tried to load Tokenize but failed to load from disc, re-caching.
[ Info: Tried to load MacroTools but failed to load from disc, re-caching.
[ Info: Tried to load Parameters but failed to load from disc, re-caching.
[ Info: Tried to load LibGit2 but failed to load from disc, re-caching.
[ Info: Tried to load Roots but failed to load from disc, re-caching.
[ Info: Tried to load Test but failed to load from disc, re-caching.
[ Info: Tried to load StaticArrays but failed to load from disc, re-caching.
[ Info: Tried to load DocStringExtensions but failed to load from disc, re-caching.
[ Info: Tried to load Mmap but failed to load from disc, re-caching.
[ Info: Tried to load TreeViews but failed to load from disc, re-caching.
[ Info: Tried to load JuliaVariables but failed to load from disc, re-caching.
[ Info: Tried to load RecipesBase but failed to load from disc, re-caching.
[ Info: Tried to load SuiteSparse but failed to load from disc, re-caching.
[ Info: Tried to load DiffRules but failed to load from disc, re-caching.
[ Info: Tried to load ModelingToolkit but failed to load from disc, re-caching.
[ Info: Tried to load Libdl but failed to load from disc, re-caching.
[ Info: Tried to load SharedArrays but failed to load from disc, re-caching.
[ Info: Tried to load SpecialFunctions but failed to load from disc, re-caching.
[ Info: Tried to load RecursiveArrayTools but failed to load from disc, re-caching.
[ Info: Tried to load REPL but failed to load from disc, re-caching.
[ Info: Tried to load IterativeSolvers but failed to load from disc, re-caching.
[ Info: Tried to load Unicode but failed to load from disc, re-caching.
[ Info: Tried to load OrderedCollections but failed to load from disc, re-caching.
[ Info: Tried to load MLStyle but failed to load from disc, re-caching.
[ Info: Tried to load Requires but failed to load from disc, re-caching.
[ Info: Tried to load CSTParser but failed to load from disc, re-caching.
[ Info: Tried to load SHA but failed to load from disc, re-caching.
[ Info: Tried to load InteractiveUtils but failed to load from disc, re-caching.
[ Info: Tried to load Markdown but failed to load from disc, re-caching.
[ Info: Tried to load GeneralizedGenerated but failed to load from disc, re-caching.
[ Info: Tried to load Distributed but failed to load from disc, re-caching.
[ Info: Tried to load FunctionWrappers but failed to load from disc, re-caching.
[ Info: Tried to load UUIDs but failed to load from disc, re-caching.
[ Info: Tried to load Serialization but failed to load from disc, re-caching.
[ Info: Tried to load NameResolution but failed to load from disc, re-caching.
[ Info: Tried to load LegibleLambdas but failed to load from disc, re-caching.
[ Info: Tried to load BinaryProvider but failed to load from disc, re-caching.
[ Info: Tried to load Statistics but failed to load from disc, re-caching.
[ Info: Tried to load Base64 but failed to load from disc, re-caching.
[ Info: Tried to load RecursiveFactorization but failed to load from disc, re-caching.
[ Info: Tried to load Logging but failed to load from disc, re-caching.
[ Info: Tried to load SparseArrays but failed to load from disc, re-caching.
[ Info: Tried to load MuladdMacro but failed to load from disc, re-caching.
[ Info: Tried to load NaNMath but failed to load from disc, re-caching.
[ Info: Tried to load Random but failed to load from disc, re-caching.
[ Info: Tried to load DelimitedFiles but failed to load from disc, re-caching.
[ Info: Tried to load DataStructures but failed to load from disc, re-caching.
[ Info: Tried to load OffsetArrays but failed to load from disc, re-caching.
[ Info: Tried to load Interpolations but failed to load from disc, re-caching.
[ Info: Tried to load BinDeps but failed to load from disc, re-caching.
[ Info: Tried to load Unitful but failed to load from disc, re-caching.
[ Info: Tried to load Compat but failed to load from disc, re-caching.
[ Info: Tried to load URIParser but failed to load from disc, re-caching.
[ Info: Tried to load CanonicalTraits but failed to load from disc, re-caching.
[ Info: Tried to load AxisAlgorithms but failed to load from disc, re-caching.
[ Info: Tried to load DiffEqBase but failed to load from disc, re-caching.
[ Info: Tried to load ArrayInterface but failed to load from disc, re-caching.
[ Info: Tried to load LinearAlgebra but failed to load from disc, re-caching.
[ Info: Tried to load Pkg but failed to load from disc, re-caching.
[ Info: Tried to load Printf but failed to load from disc, re-caching.
[ Info: Tried to load DiffEqDiffTools but failed to load from disc, re-caching.
[ Info: Tried to load UnitfulUS but failed to load from disc, re-caching.
[ Info: Tried to load TableTraits but failed to load from disc, re-caching.
[ Info: Tried to load IteratorInterfaceExtensions but failed to load from disc, re-caching.
[ Info: Tried to load Ratios but failed to load from disc, re-caching.
[ Info: Tried to load Sockets but failed to load from disc, re-caching.
[ Info: Tried to load Dates but failed to load from disc, re-caching.
[ Info: Tried to load Tokenize but failed to load from disc, re-caching.
[ Info: Tried to load MacroTools but failed to load from disc, re-caching.
[ Info: Tried to load Parameters but failed to load from disc, re-caching.
[ Info: Tried to load LibGit2 but failed to load from disc, re-caching.
[ Info: Tried to load Roots but failed to load from disc, re-caching.
[ Info: Tried to load Test but failed to load from disc, re-caching.
[ Info: Tried to load StaticArrays but failed to load from disc, re-caching.
[ Info: Tried to load DocStringExtensions but failed to load from disc, re-caching.
[ Info: Tried to load WoodburyMatrices but failed to load from disc, re-caching.
[ Info: Tried to load Mmap but failed to load from disc, re-caching.
[ Info: Tried to load TreeViews but failed to load from disc, re-caching.
[ Info: Tried to load JuliaVariables but failed to load from disc, re-caching.
[ Info: Tried to load RecipesBase but failed to load from disc, re-caching.
[ Info: Tried to load SuiteSparse but failed to load from disc, re-caching.
[ Info: Tried to load DiffRules but failed to load from disc, re-caching.
[ Info: Tried to load ModelingToolkit but failed to load from disc, re-caching.
[ Info: Tried to load Libdl but failed to load from disc, re-caching.
[ Info: Tried to load SharedArrays but failed to load from disc, re-caching.
[ Info: Tried to load SpecialFunctions but failed to load from disc, re-caching.
[ Info: Tried to load RecursiveArrayTools but failed to load from disc, re-caching.
[ Info: Tried to load REPL but failed to load from disc, re-caching.
[ Info: Tried to load IterativeSolvers but failed to load from disc, re-caching.
[ Info: Tried to load Unicode but failed to load from disc, re-caching.
[ Info: Tried to load OrderedCollections but failed to load from disc, re-caching.
[ Info: Tried to load MLStyle but failed to load from disc, re-caching.
[ Info: Tried to load Requires but failed to load from disc, re-caching.
[ Info: Tried to load CSTParser but failed to load from disc, re-caching.
[ Info: Tried to load SHA but failed to load from disc, re-caching.
[ Info: Tried to load InteractiveUtils but failed to load from disc, re-caching.
[ Info: Tried to load Markdown but failed to load from disc, re-caching.
[ Info: Tried to load GeneralizedGenerated but failed to load from disc, re-caching.
[ Info: Tried to load Distributed but failed to load from disc, re-caching.
[ Info: Tried to load FunctionWrappers but failed to load from disc, re-caching.
[ Info: Tried to load UUIDs but failed to load from disc, re-caching.
[ Info: Tried to load Serialization but failed to load from disc, re-caching.
[ Info: Tried to load NameResolution but failed to load from disc, re-caching.
[ Info: Tried to load LegibleLambdas but failed to load from disc, re-caching.
[ Info: Tried to load BinaryProvider but failed to load from disc, re-caching.
[ Info: Tried to load Statistics but failed to load from disc, re-caching.
[ Info: Tried to load Base64 but failed to load from disc, re-caching.
[ Info: Tried to load RecursiveFactorization but failed to load from disc, re-caching.
[ Info: Tried to load Logging but failed to load from disc, re-caching.
[ Info: Tried to load SparseArrays but failed to load from disc, re-caching.
[ Info: Tried to load MuladdMacro but failed to load from disc, re-caching.
[ Info: Tried to load NaNMath but failed to load from disc, re-caching.
[ Info: Tried to load Random but failed to load from disc, re-caching.
[ Info: Tried to load DelimitedFiles but failed to load from disc, re-caching.
[ Info: Tried to load DataStructures but failed to load from disc, re-caching.
[ Info: Tried to load OffsetArrays but failed to load from disc, re-caching.
[ Info: Tried to load Interpolations but failed to load from disc, re-caching.
[ Info: Tried to load BinDeps but failed to load from disc, re-caching.
[ Info: Tried to load Unitful but failed to load from disc, re-caching.
[ Info: Tried to load Compat but failed to load from disc, re-caching.
[ Info: Tried to load URIParser but failed to load from disc, re-caching.
[ Info: Tried to load CanonicalTraits but failed to load from disc, re-caching.
[ Info: Tried to load AxisAlgorithms but failed to load from disc, re-caching.
[ Info: Tried to load DiffEqBase but failed to load from disc, re-caching.
[ Info: Tried to load ArrayInterface but failed to load from disc, re-caching.
[ Info: Tried to load LinearAlgebra but failed to load from disc, re-caching.
[ Info: Tried to load Pkg but failed to load from disc, re-caching.
[ Info: Tried to load Printf but failed to load from disc, re-caching.
[ Info: Tried to load DiffEqDiffTools but failed to load from disc, re-caching.
[ Info: Tried to load UnitfulUS but failed to load from disc, re-caching.
[ Info: Tried to load TableTraits but failed to load from disc, re-caching.
[ Info: Tried to load IteratorInterfaceExtensions but failed to load from disc, re-caching.
[ Info: Tried to load Ratios but failed to load from disc, re-caching.
[ Info: Tried to load Sockets but failed to load from disc, re-caching.
[ Info: Tried to load Dates but failed to load from disc, re-caching.
[ Info: Tried to load Tokenize but failed to load from disc, re-caching.
[ Info: Tried to load MacroTools but failed to load from disc, re-caching.
[ Info: Tried to load Parameters but failed to load from disc, re-caching.
[ Info: Tried to load LibGit2 but failed to load from disc, re-caching.
[ Info: Tried to load Roots but failed to load from disc, re-caching.
[ Info: Tried to load Test but failed to load from disc, re-caching.
[ Info: Tried to load StaticArrays but failed to load from disc, re-caching.
[ Info: Tried to load DocStringExtensions but failed to load from disc, re-caching.
[ Info: Tried to load WoodburyMatrices but failed to load from disc, re-caching.
[ Info: Tried to load Mmap but failed to load from disc, re-caching.
[ Info: Tried to load TreeViews but failed to load from disc, re-caching.
[ Info: Tried to load JuliaVariables but failed to load from disc, re-caching.
[ Info: Tried to load RecipesBase but failed to load from disc, re-caching.
[ Info: Tried to load SuiteSparse but failed to load from disc, re-caching.
[ Info: Tried to load DiffRules but failed to load from disc, re-caching.
[ Info: Tried to load ModelingToolkit but failed to load from disc, re-caching.
[ Info: Tried to load Libdl but failed to load from disc, re-caching.
[ Info: Tried to load SharedArrays but failed to load from disc, re-caching.
[ Info: Tried to load SpecialFunctions but failed to load from disc, re-caching.
[ Info: Tried to load RecursiveArrayTools but failed to load from disc, re-caching.
[ Info: Tried to load REPL but failed to load from disc, re-caching.
[ Info: Tried to load IterativeSolvers but failed to load from disc, re-caching.
[ Info: Tried to load Unicode but failed to load from disc, re-caching.
[ Info: Tried to load OrderedCollections but failed to load from disc, re-caching.
[ Info: Tried to load MLStyle but failed to load from disc, re-caching.
[ Info: Tried to load Requires but failed to load from disc, re-caching.
[ Info: Tried to load CSTParser but failed to load from disc, re-caching.
[ Info: Tried to load SHA but failed to load from disc, re-caching.
[ Info: Tried to load InteractiveUtils but failed to load from disc, re-caching.
[ Info: Tried to load Markdown but failed to load from disc, re-caching.
[ Info: Tried to load GeneralizedGenerated but failed to load from disc, re-caching.
[ Info: Tried to load Distributed but failed to load from disc, re-caching.
[ Info: Tried to load FunctionWrappers but failed to load from disc, re-caching.
[ Info: Tried to load UUIDs but failed to load from disc, re-caching.
[ Info: Tried to load Serialization but failed to load from disc, re-caching.
[ Info: Tried to load NameResolution but failed to load from disc, re-caching.
[ Info: Tried to load LegibleLambdas but failed to load from disc, re-caching.
[ Info: Tried to load BinaryProvider but failed to load from disc, re-caching.
[ Info: Tried to load Statistics but failed to load from disc, re-caching.
[ Info: Tried to load Base64 but failed to load from disc, re-caching.
[ Info: Tried to load RecursiveFactorization but failed to load from disc, re-caching.
[ Info: Tried to load Logging but failed to load from disc, re-caching.
[ Info: Tried to load SparseArrays but failed to load from disc, re-caching.
[ Info: Tried to load MuladdMacro but failed to load from disc, re-caching.
[ Info: Tried to load NaNMath but failed to load from disc, re-caching.
[ Info: Tried to load Random but failed to load from disc, re-caching.
[ Info: Tried to load DelimitedFiles but failed to load from disc, re-caching.
[ Info: Tried to load DataStructures but failed to load from disc, re-caching.
[ Info: Tried to load OffsetArrays but failed to load from disc, re-caching.
[ Info: Tried to load Interpolations but failed to load from disc, re-caching.
[ Info: Tried to load BinDeps but failed to load from disc, re-caching.
[ Info: Tried to load Unitful but failed to load from disc, re-caching.
[ Info: Tried to load Compat but failed to load from disc, re-caching.
[ Info: Tried to load URIParser but failed to load from disc, re-caching.
[ Info: Tried to load CanonicalTraits but failed to load from disc, re-caching.
[ Info: Tried to load AxisAlgorithms but failed to load from disc, re-caching.
[ Info: Tried to load DiffEqBase but failed to load from disc, re-caching.
[ Info: Tried to load ArrayInterface but failed to load from disc, re-caching.
[ Info: Tried to load LinearAlgebra but failed to load from disc, re-caching.
[ Info: Tried to load Pkg but failed to load from disc, re-caching.
[ Info: Tried to load Printf but failed to load from disc, re-caching.
[ Info: Tried to load DiffEqDiffTools but failed to load from disc, re-caching.
[ Info: Tried to load UnitfulUS but failed to load from disc, re-caching.
[ Info: Tried to load TableTraits but failed to load from disc, re-caching.
[ Info: Tried to load IteratorInterfaceExtensions but failed to load from disc, re-caching.
[ Info: Tried to load Ratios but failed to load from disc, re-caching.
[ Info: Tried to load Sockets but failed to load from disc, re-caching.
[ Info: Tried to load Dates but failed to load from disc, re-caching.
[ Info: Tried to load Tokenize but failed to load from disc, re-caching.
[ Info: Tried to load MacroTools but failed to load from disc, re-caching.
[ Info: Tried to load Parameters but failed to load from disc, re-caching.
[ Info: Tried to load LibGit2 but failed to load from disc, re-caching.
[ Info: Tried to load Roots but failed to load from disc, re-caching.
[ Info: Tried to load Test but failed to load from disc, re-caching.
[ Info: Tried to load StaticArrays but failed to load from disc, re-caching.
[ Info: Tried to load DocStringExtensions but failed to load from disc, re-caching.
[ Info: Tried to load WoodburyMatrices but failed to load from disc, re-caching.
[ Info: Tried to load Mmap but failed to load from disc, re-caching.
[ Info: Tried to load TreeViews but failed to load from disc, re-caching.
[ Info: Tried to load JuliaVariables but failed to load from disc, re-caching.
[ Info: Tried to load RecipesBase but failed to load from disc, re-caching.
[ Info: Tried to load SuiteSparse but failed to load from disc, re-caching.
[ Info: Tried to load DiffRules but failed to load from disc, re-caching.
[ Info: Tried to load ModelingToolkit but failed to load from disc, re-caching.
[ Info: Tried to load Libdl but failed to load from disc, re-caching.
[ Info: Tried to load SharedArrays but failed to load from disc, re-caching.
[ Info: Tried to load SpecialFunctions but failed to load from disc, re-caching.
[ Info: Tried to load RecursiveArrayTools but failed to load from disc, re-caching.
[ Info: Tried to load REPL but failed to load from disc, re-caching.
```

Versions:

```julia
(eglot-jl) pkg> st -m
    Status `~/.emacs.d/straight/repos/eglot-jl/Manifest.toml`
  [00ebfdb7] ~ CSTParser v1.0.0 ⇒ v1.0.0 ⚲
  [ffa9a821]   DocumentFormat v1.0.1
  [682c06a0]   JSON v0.21.0
  [2b0e0bc5] ↑ LanguageServer v0.6.1-DEV #f31248d (https://github.com/julia-vscode/LanguageServer.jl.git) ⚲ ⇒ v1.0.1-DEV #d9772fb (https://github.com/julia-vscode/LanguageServer.jl.git)
  [69de0a69] ↑ Parsers v0.3.8 ⇒ v0.3.10
  [b3cc710f]   StaticLint v1.0.2
  [cf896787]   SymbolServer v1.0.1
  [0796e94c] ↑ Tokenize v0.5.6 ⇒ v0.5.7
  [30578b45]   URIParser v0.4.0
  [2a0f44e3]   Base64 
  [ade2ca70]   Dates 
  [8ba89e20]   Distributed 
  [b77e0a4c]   InteractiveUtils 
  [76f85450]   LibGit2 
  [56ddb016]   Logging 
  [d6f4376e]   Markdown 
  [a63ad114]   Mmap 
  [44cfe95a]   Pkg 
  [de0858da]   Printf 
  [3fa0cd96]   REPL 
  [9a3f8284]   Random 
  [ea8e919c]   SHA 
  [9e88b42a]   Serialization 
  [6462fe0b]   Sockets 
  [8dfed614]   Test 
  [cf7118a7]   UUIDs 
  [4ec0a83e]   Unicode 

julia> versioninfo()
Julia Version 1.2.0
Commit c6da87ff4b (2019-08-20 00:03 UTC)
Platform Info:
  OS: Linux (x86_64-unknown-linux-gnu)
  CPU: Intel(R) Core(TM) i5-5300U CPU @ 2.30GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.1 (ORCJIT, broadwell)
```

**Comment by non-Jedi** [comment]
I can't reproduce this using SymbolServer directly. If I run the following, it works as expected. Something to do with its integration with LanguageServer.jl?

```julia
using SymbolServer
ssp = SymbolServerProcess(environment="/home/adam/repos/ChemicalEngineeringToolbox.jl/")
packages = SymbolServer.getstore(ssp)
```

**Comment by non-Jedi** [comment]
Here's what I know:

1. If I create a `LanguageServerInstance` (with `depot_path=""`) and `run` it, this bug shows up.
2. If I <kbd><kbd>Ctrl</kbd>-<kbd>c</kbd></kbd> the running/looping `LanguageServerInstance`, create a new one, and `run` it, this bug shows up.
3. If I <kbd><kbd>Ctrl</kbd>-<kbd>c</kbd></kbd> the running/looping `LanguageServerInstance`, then follow below steps, `getstore` returns as expected.
  1. `using LanguageServer.SymbolServer`
  2. `ssp = SymbolServerProcess(depot="", environment="/home/adam/repos/ChemicalEngineeringToolbox.jl/")`
  3. `packages = getstore(ssp)`
4. If after step 3, I then create a new `LanguageServerInstance` and `run` it, the server runs fine without looping on re-caching.

All of this makes no sense to me as I'm doing exactly the same thing as LanguageServer.jl with `SymbolServerProcess`, but it works if I do so manually.

**Comment by non-Jedi** [comment]
It seems to be the bare `using LanguageServer.SymbolServer` that fixes it rather than invoking `getstore` manually. Still very odd. Probably the reason you haven't seen this bug in vscode is because you actually have `using SymbolServer` in your script even though it isn't used directly:

https://github.com/julia-vscode/julia-vscode/blob/8f95952314135698127bb6ebb4031862b74cde33/scripts/languageserver/main.jl#L21

**Comment by non-Jedi** [comment]
This might be a duplicate of #54.

**Comment by sbromberger** [comment]
Jumping in here in the hope that this helps: I can reproduce this in a pretty clean Julia environment by executing
```
server = SymbolServerProcess()
load_project_packages(server)
```

That is, I get 
```
...
[ Info: DelimitedFiles not stored on disc                                                                                                                                                                                   [161/1872]
[ Info: Tried to load LibGit2 but failed to load from disc, re-caching.
[ Info: LightGraphs not stored on disc
[ Info: Mmap not stored on disc
[ Info: Test not stored on disc
[ Info: Tried to load REPL but failed to load from disc, re-caching.
[ Info: ArnoldiMethod not stored on disc
[ Info: Inflate not stored on disc
[ Info: Libdl not stored on disc
[ Info: InteractiveUtils not stored on disc
[ Info: LibGit2 not stored on disc
[ Info: Tried to load REPL but failed to load from disc, re-caching.
[ Info: SimpleTraits not stored on disc
[ Info: Tried to load REPL but failed to load from disc, re-caching.
[ Info: Tried to load Distributed but failed to load from disc, re-caching.
[ Info: Tried to load Distributed but failed to load from disc, re-caching.
[ Info: Tried to load LightGraphs but failed to load from disc, re-caching.                                                                                                                                                           [ Info: Tried to load StaticArrays but failed to load from disc, re-caching.
[ Info: Tried to load SharedArrays but failed to load from disc, re-caching.
[ Info: Tried to load LightGraphs but failed to load from disc, re-caching.
[ Info: StaticArrays not stored on disc
[ Info: SharedArrays not stored on disc
[ Info: Tried to load LightGraphs but failed to load from disc, re-caching.
[ Info: Tried to load LinearAlgebra but failed to load from disc, re-caching.
...
```

in an endless stream.

**Comment by kdheepak** [comment]
I'm also getting the same issue as this. Surprisingly, even with a very minimal environment `Project.toml`:

```
[deps]
LanguageServer = "2b0e0bc5-e4fd-59b4-8912-456d1b03d8d7"
OhMyREPL = "5fb14364-9ced-5910-84b2-373655c76a03"
Revise = "295af30f-e4ad-537b-8983-00126c2a3abe"
```

I still get an endless stream.

**Comment by kdheepak** [comment]
I'm able to use LanguageServer in some projects with the following configuration steps. I don't fully understand why this works though, perhaps someone can shed some light. 

[This snippet from my .vimrc is how I start `LanguageServer`](https://github.com/kdheepak/dotfiles/blob/7a6bf47ae1c56115116a99763bc5267599f569f2/.vimrc#L755-L762):

```vimrc
let g:LanguageClient_serverCommands = {
    \ 'julia': ['~/Applications/Julia-1.3.app/Contents/Resources/julia/bin/julia', '--startup-file=no', '--history-file=no', '-e', '
    \   using LanguageServer;
    \   p = joinpath(dirname(dirname(pathof(LanguageServer))), "contrib", "languageserver.sh");
    \   run(`$p`);
    \   '],
    \ 'python': ['~/miniconda3/bin/pyls'],
    \ }
```

I had to change a line in `languageserver.sh` to the following:

```julia
$JULIABIN --project --startup-file=no --history-file=no -e \
    "using LanguageServer; using Pkg; server = LanguageServer.LanguageServerInstance(stdin, stdout, $DEBUG, dirname(Pkg.Types.Context().env.project_file)); server.runlinter = true; run(server);" \
    <&0 >&1 &
```

i.e., 

1) Add `--project` to command line arguments of Julia
2) Remove `import SymbolServer` (Or run `add SymbolServer` in Julia)
3) Add `using Pkg;` to the executable string and add `dirname(Pkg.Types.Context().env.project_file)` as the fourth argument to `LanguageServerInstance`.

I tried not using `--project` and/or not using `dirname(Pkg.Types.Context().env.project_file)`, and in both cases I always got the infinite recaching issue since my root environment `Manifest.toml` is being used for the list of packages to cache. Only adding _both_ `--project` and `dirname(Pkg.Types.Context().env.project_file)` and starting the script from the root directory of a repository results in using the `Manifest.toml` file of that repository for list of packages to cache. I think all this does is just minimize the chances that the infinite recaching issue will happen.

**Comment by davidanthoff** [comment]
`master` is more or less a rewrite of the entire symbol server, so I'm going to close this for now. If this is still a problem, please reopen!

**Comment by sbromberger** [comment]
I don't know how to make the latest master work:
```
(v1.3) pkg> dev StaticLint
  Updating registry at `~/.julia/registries/General`
  Updating git-repo `https://github.com/JuliaRegistries/General.git`
  Updating git-repo `https://github.com/julia-vscode/StaticLint.jl.git`
 Resolving package versions...
ERROR: Unsatisfiable requirements detected for package SymbolServer [cf896787]:
 SymbolServer [cf896787] log:
 ├─possible versions are: [0.0.1, 0.1.0-0.1.3, 0.2.0-0.2.5, 1.0.0-1.0.1, 2.0.0] or uninstalled
 ├─restricted to versions 2 by StaticLint [b3cc710f], leaving only versions 2.0.0
 │ └─StaticLint [b3cc710f] log:
 │   ├─possible versions are: 3.0.1 or uninstalled
 │   └─StaticLint [b3cc710f] is fixed to version 3.0.1-DEV
 └─restricted to versions 1.0.1 by an explicit requirement — no versions left
```

**Comment by davidanthoff** [comment]
You certainly can't, we need to update LS to work with this version here before it is usable. Once we have a set of packages that works together nicely, I'll tag a whole round of new releases.
